pkg/controllers/core/deployer_controller.go

```diff
@@ -132,8 +137,18 @@ func (r *DeployerReconciler) reconcile(ctx context.Context, log logr.Logger, dep
 	deployer.Status.ServiceName = childService.Name
 	deployer.Status.PropagateServiceStatus(&childService.Status)
 
-	deployer.Status.ObservedGeneration = deployer.Generation
+	// reconcile ingress
+	childIngress, err := r.reconcileIngress(ctx, log, deployer, childService.Name)
+	if err != nil {
+		log.Error(err, "unable to reconcile Ingress", "deployer", deployer)
+		return ctrl.Result{}, err
+	}
+	if childIngress != nil {
```

Comment:
Also need to update the status is the ingress is nil. If for some reason the ingress previously existed and now no longer needs to exists, that change should be reflected on the Deployer's status.

Reply:
Updated the status to False with helpful msg & reason

Reply:
Take a look at https://github.com/projectriff/system/blob/8c95a169d0f915e5d2f2796127ddb28a214f9cd0/pkg/controllers/build/application_controller.go#L114-L123